api/nakadi-event-bus-api.yaml

```diff
@@ -606,22 +488,388 @@ paths:
+        and the description of commit endpoint.
+      parameters:
+        - $ref: '#/parameters/SubscriptionId'
+        - name: commit_timeout
+          in: query
+          description: |
+            The amount of time allowed between commits. If no commit is performed within this amount of time
+            - the connection will be closed from Nakadi side. Specified in seconds.
+          type: integer
+          format: int32
+          required: false
+          default: 30
+        - name: window_size
+          in: query
+          description: |
+            The amount of uncommitted events Nakadi will stream before pausing the stream. When in paused
+            state and commit comes - the stream will continue. Minimal window size is 1.
+          type: integer
+          format: int32
+          required: false
+          default: 10000
+        - name: commit_mode
+          in: query
+          description: |
+            The commit mode that will be used for this stream. Currently only `commit_at_will` is supported.
+            Possible values:
+
+            - `commit_at_will`: client is free to choose how often he wants to commit. Of course the commit_timeout
+            property should still be respected.
+            When a batch is committed that also automatically commits all previous batches that were sent in a stream
+            (if the commit_semantics parameter is `per_partition` - that will commit all previous batches only for
+            current partition).
+
+            - `commit_each`: this commit mode requires each batch to be committed, and it should be done in correct
+            order. Committing a cursor when the previous is not yet committed is not possible.
+
+            - `auto_commit`: everything that Nakadi sends to stream is considered to be committed. No actions are
+            required from client side to commit. Be aware that with this commit mode loosing of some events is very
+            probable. So you should know what you are doing if you use autocommit.
+          type: string
+          enum:
+            - commit_at_will
+            - commit_each
+            - auto_commit
+          required: false
+          default: commit_at_will
+        - name: commit_semantics
+          in: query
           description: |
-            The flow id of the request, which is written into the logs and passed to called
-            services. Helpful for operational troubleshooting and log analysis.
+            The semantics of commit that will be used for this stream. Possible values:
+
+            - `per_partition`: Each partition is committed separatelly. Each cursor in a stream belongs to one of the
+            partitions of that stream; committing that cursor will commit the data only for that partition.
+
+            - `per_stream`: When you commit a cursor - all data in a stream before this cursor is
+            commited (doesn't matter to which partition it belongs)
```

Comment:
@rcillo We are moving there because users who have to consume data think of stream as a stream, but not as a several streams.
And as far as all described their positions, I'll to the same: personally I think that we must get rid of this parameter, but I tend not to multiplex cursors in one stream, but use megacursor for all sent data, so particular consumer must not think of the fact that several streams are multiplexed in one.

Probably that is why we have this parameter - someone (as far as I understand, @jkliff knows who) had problems with understanding of this multiplexity.